nixos/modules/services/web-apps/librechat.nix

```diff
@@ -0,0 +1,162 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.librechat;
+  format = pkgs.formats.yaml { };
+  configFile = format.generate "librechat.yaml" cfg.settings;
+  # Thanks to https://github.com/nix-community/home-manager/blob/60e4624302d956fe94d3f7d96a560d14d70591b9/modules/lib/shell.nix :)
+  export = n: v: ''export ${n}="${builtins.toString v}"'';
+  exportAll = vars: lib.concatStringsSep "\n" (lib.mapAttrsToList export vars);
+  exportCredentials = n: _: ''export ${n}="$(${pkgs.systemd}/bin/systemd-creds cat ${n}_FILE)"'';
+  exportAllCredentials = vars: lib.concatStringsSep "\n" (lib.mapAttrsToList exportCredentials vars);
+  transformCredential = n: v: "${n}_FILE:${v}";
+  getLoadCredentialList = lib.mapAttrsToList transformCredential cfg.credentials;
+in
+{
+  options.services.librechat = {
+    enable = lib.mkEnableOption "Whether to enable the LibreChat server.";
+    package = lib.mkPackageOption pkgs "librechat" { };
+    openFirewall = lib.mkEnableOption "Whether to open the port in the firewall.";
+    logDir = lib.mkOption {
+      type = lib.types.str;
+      default = "/var/lib/librechat";
+      example = "/persist/librechat";
+      description = "Absolute path for where the LibreChat server will use as its log directory.";
+    };
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 3080;
+      example = 2309;
+      description = "The value that will be passed to the PORT environment variable, telling LibreChat what to listen on.";
+    };
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = "librechat";
+      example = "alice";
+      description = "The user to run the service as.";
+    };
+    group = lib.mkOption {
+      type = lib.types.str;
+      default = "librechat";
+      example = "users";
+      description = "The group to run the service as.";
+    };
+    credentials = lib.mkOption {
+      type =
+        with lib.types;
+        submodule {
+          freeformType = attrsOf path;
+        };
```

Comment:
Is there any reason this should be a submodule? Just an `attrsOf path` makes far more sense IMO